tests/unit_tests/test_clans.py

```diff
@@ -0,0 +1,168 @@
+        (447, 'Dragonfire', '', 'a'),
+        (449, 'Blackheart', '', 'b'),
+        (448, 'machina', '', 'c'),
+        (453, 'Kammer', '', 'd'),
+        (450, 'reddev32', '', 'e'),
+        (457, 'VoR_Tex', '', 'f'),
+        (473, 'Koecher', '', 'g'),
+        (474, 'Pathogen', '', 'h'),
+        (479, 'Stromfresser', '', 'i')""")
+
+    def finalizer():
+        with db.connection:
+            cursor = db.connection.cursor()
+            cursor.execute("DELETE FROM login")
+
+    request.addfinalizer(finalizer)
+
+def test_clan_list(test_client, clans, clan_members, clan_login):
+    response = test_client.get('/clans')
+
+    assert response.status_code == 200
+    assert response.content_type == 'application/vnd.api+json'
+
+    result = json.loads(response.data.decode('utf-8'))
+    assert 'data' in result
+    assert len(result['data']) == 9
+
+    for item in result['data']:
+        assert 'clan_id' in item
+        assert 'status' in item
+        assert 'clan_name' in item
+        assert 'clan_tag' in item
+        assert 'clan_leader_id' in item
+        assert 'clan_founder_id' in item
+        assert 'clan_desc' in item
+        assert 'create_date' in item
+        assert 'leader_name' in item
+        assert 'founder_name' in item
+
+    assert result['data'][0]['member_count'] == 3
+
+def test_clan_founder_names(test_client, clans, clan_login):
+    response = test_client.get('/clans')
+    result = json.loads(response.data.decode('utf-8'))
+    assert len(result['data']) == 9
+    assert result['data'][0]['founder_name'] == 'Dragonfire'
+    assert result['data'][1]['founder_name'] == 'Blackheart'
+    assert result['data'][2]['founder_name'] == 'reddev32'
+    assert result['data'][3]['founder_name'] == 'VoR_Tex'
+    assert result['data'][4]['founder_name'] == 'machina'
+    assert result['data'][5]['founder_name'] == 'Koecher'
+    assert result['data'][6]['founder_name'] == 'Pathogen'
+    assert result['data'][7]['founder_name'] == 'Kammer'
+    assert result['data'][8]['founder_name'] == 'Dragonfire'
+
+def test_clan_leader_names(test_client, clans, clan_login):
+    response = test_client.get('/clans')
+    result = json.loads(response.data.decode('utf-8'))
+    assert len(result['data']) == 9
+    assert result['data'][0]['leader_name'] == 'Dragonfire'
+    assert result['data'][1]['leader_name'] == 'Blackheart'
+    assert result['data'][2]['leader_name'] == 'reddev32'
+    assert result['data'][3]['leader_name'] == 'VoR_Tex'
+    assert result['data'][4]['leader_name'] == 'Dragonfire'
+    assert result['data'][5]['leader_name'] == 'Koecher'
+    assert result['data'][6]['leader_name'] == 'Pathogen'
+    assert result['data'][7]['leader_name'] == 'Kammer'
+    assert result['data'][8]['leader_name'] == 'Stromfresser'
+
+def test_invalid_clan(test_client):
+    response = test_client.get('/clan/42')
+
+    # TODO: is this correct, or should we return 404?
```

Comment:
As far as I understand jsonapi, 404 (http://jsonapi.org/format/#fetching-resources)

This isn't the case yet in other implementations either